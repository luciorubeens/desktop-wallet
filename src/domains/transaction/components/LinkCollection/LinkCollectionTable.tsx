import { Button } from "app/components/Button";
import { Icon } from "app/components/Icon";
import { RadioButton } from "app/components/RadioButton";
import { Table } from "app/components/Table";
import React from "react";
import { ValidationRules } from "react-hook-form";
import { useTranslation } from "react-i18next";

import { EntityLink } from "./LinkCollection.models";

type Props = {
	name: string;
	fields: EntityLink[];
	remove: (index: number) => void;
	registerRef: (options?: ValidationRules) => (ref: HTMLInputElement | null) => void;

	optionsTypes?: string[];
	optionsTypesTitle?: string;
	onOptionChange?: (link: EntityLink) => void;
	optionChecked?: number;
};

export const LinkCollectionTable = ({
	remove,
	optionsTypesTitle,
	optionsTypes,
	registerRef,
	fields,
	optionChecked,
	onOptionChange,
	name,
}: Props) => {
	const { t } = useTranslation();

	const columns = [];

	if (optionsTypesTitle) {
		columns.push({
			Header: optionsTypesTitle,
			accessor: "isSelected",
		});
	}

	columns.push(
		{
			Header: t("COMMON.NAME"),
			accessor: "type",
		},
		{
			Header: t("COMMON.LINK"),
			accessor: "link",
		},
		{
			Header: " ",
		},
	);

	return (
		<div data-testid="LinkCollectionTable">
			<Table columns={columns} data={fields}>
				{(rowData: EntityLink, rowIndex: any) => (
					<tr
						data-testid="LinkCollectionTable__row"
						key={rowData.value}
						className="font-semibold border-b border-theme-neutral-200"
					>
						{optionsTypesTitle && (
							<td className="w-16 text-center">
								{optionsTypes?.includes(rowData.type) && (
									<RadioButton
										data-testid="LinkCollectionTable__row__option"
										checked={optionChecked === rowIndex}
										onChange={() => onOptionChange?.(rowData)}
									/>
								)}
							</td>
						)}

						<td
							data-testid="LinkCollectionTable__row__type"
							className={`w-40 ${rowIndex > 0 ? "py-6" : "pb-6 pt-2"}`}
						>
							<input
								className="hidden"
								name={`${name}[${rowIndex}].type`}
								ref={registerRef()}
								defaultValue={rowData.type}
								readOnly
							/>
							<span>{t(`TRANSACTION.LINK_TYPES.${rowData.type.toUpperCase()}`)}</span>
						</td>

						<td
							data-testid="LinkCollectionTable__row__value"
							className={rowIndex > 0 ? "py-6" : "pb-6 pt-2"}
						>
							<input
								className="focus:outline-none"
								name={`${name}[${rowIndex}].value`}
								ref={registerRef()}
								defaultValue={rowData.value}
								readOnly
							/>
						</td>

						<td className={`w-16 text-right ${rowIndex === 0 && "pb-4"}`}>
							<Button
								data-testid="LinkCollectionTable__remove-link"
								size="icon"
								variant="plain"
								onClick={() => remove(rowIndex)}
							>
								<Icon name="Trash" />
							</Button>
						</td>
					</tr>
				)}
			</Table>
		</div>
	);
};
