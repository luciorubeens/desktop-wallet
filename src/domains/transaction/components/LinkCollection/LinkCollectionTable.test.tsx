import React from "react";
import { useFieldArray, useForm } from "react-hook-form";
import { act, fireEvent, render } from "utils/testing-library";

import { LinkCollectionTable } from "./LinkCollectionTable";

describe("LinkCollectionTable", () => {
	const options = [
		{
			type: "facebook",
			value: "https://facebook.com",
		},
		{
			type: "instagram",
			value: "https://instagram.com",
		},
		{
			type: "twitter",
			value: "https://twitter.com",
		},
	];

	it("should display default links", () => {
		const Component = () => {
			const name = "example";
			const { control, register } = useForm({
				defaultValues: {
					example: options,
				},
				mode: "onChange",
			});

			const { fields, remove } = useFieldArray({
				control,
				name,
				keyName: "value",
			});

			return <LinkCollectionTable remove={remove} registerRef={register} fields={fields as any} name={name} />;
		};

		const { getAllByTestId, asFragment } = render(<Component />);

		expect(getAllByTestId("LinkCollectionTable__row")).toHaveLength(3);
		expect(asFragment()).toMatchSnapshot();
	});

	it("should delete link from DOM and form", () => {
		let currentValues: any[] = [];

		const Component = () => {
			const name = "example";
			const { control, register, watch } = useForm({
				defaultValues: {
					example: options,
				},
				mode: "onChange",
			});
			currentValues = watch(name);

			const { fields, remove } = useFieldArray({
				control,
				name,
				keyName: "value",
			});

			return <LinkCollectionTable remove={remove} registerRef={register} fields={fields as any} name={name} />;
		};

		const { getAllByTestId, asFragment } = render(<Component />);

		expect(getAllByTestId("LinkCollectionTable__row")).toHaveLength(3);

		act(() => {
			fireEvent.click(getAllByTestId("LinkCollectionTable__remove-link")[0]);
		});

		expect(getAllByTestId("LinkCollectionTable__row")).toHaveLength(2);
		expect(currentValues).toEqual([
			{ type: "facebook", value: "https://facebook.com" },
			{ type: "instagram", value: "https://instagram.com" },
		]);
		expect(asFragment()).toMatchSnapshot();
	});

	describe("With Options", () => {
		const Component = ({
			optionChecked,
			onOptionChange,
		}: {
			optionChecked?: number;
			onOptionChange?: (data: any) => void;
		}) => {
			const name = "example";
			const { control, register } = useForm({
				defaultValues: {
					example: options,
				},
				mode: "onChange",
			});

			const { fields, remove } = useFieldArray({
				control,
				name,
				keyName: "value",
			});

			return (
				<LinkCollectionTable
					optionsTypes={["facebook", "instagram"]}
					optionsTypesTitle="Avatar"
					optionChecked={optionChecked}
					onOptionChange={onOptionChange}
					remove={remove}
					registerRef={register}
					fields={fields as any}
					name={name}
				/>
			);
		};

		it("should show with options column", () => {
			const { getAllByTestId, asFragment } = render(<Component />);

			expect(getAllByTestId("LinkCollectionTable__row__option")).toHaveLength(2);
			expect(asFragment()).toMatchSnapshot();
		});

		it("should show checked option", () => {
			const { getAllByTestId, asFragment } = render(<Component optionChecked={1} />);

			expect(getAllByTestId("LinkCollectionTable__row__option")[0]).not.toBeChecked();
			expect(getAllByTestId("LinkCollectionTable__row__option")[1]).toBeChecked();
			expect(asFragment()).toMatchSnapshot();
		});

		it("should emit event on change", () => {
			const onOptionChange = jest.fn();
			const { getAllByTestId, asFragment } = render(
				<Component optionChecked={1} onOptionChange={onOptionChange} />,
			);

			fireEvent.click(getAllByTestId("LinkCollectionTable__row__option")[0]);

			expect(onOptionChange).toHaveBeenCalledWith({ type: "facebook", value: "https://facebook.com" });
			expect(asFragment()).toMatchSnapshot();
		});
	});
});
