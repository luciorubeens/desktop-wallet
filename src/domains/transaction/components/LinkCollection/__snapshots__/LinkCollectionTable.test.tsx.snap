// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkCollectionTable With Options should emit event on change 1`] = `
<DocumentFragment>
  <div
    data-testid="LinkCollectionTable"
  >
    <div
      class="sc-AxhCb tChBQ"
      role="table"
    >
      <table
        class="table-auto"
      >
        <thead>
          <tr
            role="row"
          >
            <th
              class="text-sm text-left select-none text-theme-neutral"
              colspan="1"
              data-testid="table__th--0"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex flex-inline align-top "
              >
                <div>
                  Avatar
                </div>
                <div
                  class="flex items-center ml-2 text-theme-color-neutral"
                  data-testid="table__Sort"
                >
                  <div
                    class="sc-AxirZ jMwzyA"
                    height="12"
                    width="15"
                  >
                    <svg>
                      sort.svg
                    </svg>
                  </div>
                </div>
              </div>
            </th>
            <th
              class="text-sm text-left select-none text-theme-neutral"
              colspan="1"
              data-testid="table__th--1"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex flex-inline align-top "
              >
                <div>
                  Name
                </div>
                <div
                  class="flex items-center ml-2 text-theme-color-neutral"
                  data-testid="table__Sort"
                >
                  <div
                    class="sc-AxirZ jMwzyA"
                    height="12"
                    width="15"
                  >
                    <svg>
                      sort.svg
                    </svg>
                  </div>
                </div>
              </div>
            </th>
            <th
              class="text-sm text-left select-none text-theme-neutral"
              colspan="1"
              data-testid="table__th--2"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex flex-inline align-top "
              >
                <div>
                  Link
                </div>
                <div
                  class="flex items-center ml-2 text-theme-color-neutral"
                  data-testid="table__Sort"
                >
                  <div
                    class="sc-AxirZ jMwzyA"
                    height="12"
                    width="15"
                  >
                    <svg>
                      sort.svg
                    </svg>
                  </div>
                </div>
              </div>
            </th>
            <th
              class="text-sm text-left select-none text-theme-neutral"
              colspan="1"
              data-testid="table__th--3"
              role="columnheader"
            >
              <div
                class="flex flex-inline align-top "
              >
                <div>
                   
                </div>
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            class="font-semibold border-b border-theme-neutral-200"
            data-testid="LinkCollectionTable__row"
          >
            <td
              class="w-16 text-center"
            >
              <input
                class="sc-AxiKw fIRSaL"
                color="success"
                data-testid="LinkCollectionTable__row__option"
                type="radio"
              />
            </td>
            <td
              class="w-40 pb-6 pt-2"
              data-testid="LinkCollectionTable__row__type"
            >
              <input
                class="hidden"
                name="example[0].type"
                readonly=""
                value="facebook"
              />
              <span>
                Facebook
              </span>
            </td>
            <td
              class="pb-6 pt-2"
              data-testid="LinkCollectionTable__row__value"
            >
              <input
                class="focus:outline-none"
                name="example[0].value"
                readonly=""
                value="https://facebook.com"
              />
            </td>
            <td
              class="w-16 text-right pb-4"
            >
              <button
                class="sc-AxjAm bBBiEi"
                color="primary"
                data-testid="LinkCollectionTable__remove-link"
                type="button"
              >
                <div
                  class="sc-AxirZ iQaTP"
                  height="1em"
                  width="1em"
                >
                  <svg>
                    trash.svg
                  </svg>
                </div>
              </button>
            </td>
          </tr>
          <tr
            class="font-semibold border-b border-theme-neutral-200"
            data-testid="LinkCollectionTable__row"
          >
            <td
              class="w-16 text-center"
            >
              <input
                checked=""
                class="sc-AxiKw fIRSaL"
                color="success"
                data-testid="LinkCollectionTable__row__option"
                type="radio"
              />
            </td>
            <td
              class="w-40 py-6"
              data-testid="LinkCollectionTable__row__type"
            >
              <input
                class="hidden"
                name="example[1].type"
                readonly=""
                value="instagram"
              />
              <span>
                Instagram
              </span>
            </td>
            <td
              class="py-6"
              data-testid="LinkCollectionTable__row__value"
            >
              <input
                class="focus:outline-none"
                name="example[1].value"
                readonly=""
                value="https://instagram.com"
              />
            </td>
            <td
              class="w-16 text-right false"
            >
              <button
                class="sc-AxjAm bBBiEi"
                color="primary"
                data-testid="LinkCollectionTable__remove-link"
                type="button"
              >
                <div
                  class="sc-AxirZ iQaTP"
                  height="1em"
                  width="1em"
                >
                  <svg>
                    trash.svg
                  </svg>
                </div>
              </button>
            </td>
          </tr>
          <tr
            class="font-semibold border-b border-theme-neutral-200"
            data-testid="LinkCollectionTable__row"
          >
            <td
              class="w-16 text-center"
            />
            <td
              class="w-40 py-6"
              data-testid="LinkCollectionTable__row__type"
            >
              <input
                class="hidden"
                name="example[2].type"
                readonly=""
                value="twitter"
              />
              <span>
                Twitter
              </span>
            </td>
            <td
              class="py-6"
              data-testid="LinkCollectionTable__row__value"
            >
              <input
                class="focus:outline-none"
                name="example[2].value"
                readonly=""
                value="https://twitter.com"
              />
            </td>
            <td
              class="w-16 text-right false"
            >
              <button
                class="sc-AxjAm bBBiEi"
                color="primary"
                data-testid="LinkCollectionTable__remove-link"
                type="button"
              >
                <div
                  class="sc-AxirZ iQaTP"
                  height="1em"
                  width="1em"
                >
                  <svg>
                    trash.svg
                  </svg>
                </div>
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LinkCollectionTable With Options should show checked option 1`] = `
<DocumentFragment>
  <div
    data-testid="LinkCollectionTable"
  >
    <div
      class="sc-AxhCb tChBQ"
      role="table"
    >
      <table
        class="table-auto"
      >
        <thead>
          <tr
            role="row"
          >
            <th
              class="text-sm text-left select-none text-theme-neutral"
              colspan="1"
              data-testid="table__th--0"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex flex-inline align-top "
              >
                <div>
                  Avatar
                </div>
                <div
                  class="flex items-center ml-2 text-theme-color-neutral"
                  data-testid="table__Sort"
                >
                  <div
                    class="sc-AxirZ jMwzyA"
                    height="12"
                    width="15"
                  >
                    <svg>
                      sort.svg
                    </svg>
                  </div>
                </div>
              </div>
            </th>
            <th
              class="text-sm text-left select-none text-theme-neutral"
              colspan="1"
              data-testid="table__th--1"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex flex-inline align-top "
              >
                <div>
                  Name
                </div>
                <div
                  class="flex items-center ml-2 text-theme-color-neutral"
                  data-testid="table__Sort"
                >
                  <div
                    class="sc-AxirZ jMwzyA"
                    height="12"
                    width="15"
                  >
                    <svg>
                      sort.svg
                    </svg>
                  </div>
                </div>
              </div>
            </th>
            <th
              class="text-sm text-left select-none text-theme-neutral"
              colspan="1"
              data-testid="table__th--2"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex flex-inline align-top "
              >
                <div>
                  Link
                </div>
                <div
                  class="flex items-center ml-2 text-theme-color-neutral"
                  data-testid="table__Sort"
                >
                  <div
                    class="sc-AxirZ jMwzyA"
                    height="12"
                    width="15"
                  >
                    <svg>
                      sort.svg
                    </svg>
                  </div>
                </div>
              </div>
            </th>
            <th
              class="text-sm text-left select-none text-theme-neutral"
              colspan="1"
              data-testid="table__th--3"
              role="columnheader"
            >
              <div
                class="flex flex-inline align-top "
              >
                <div>
                   
                </div>
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            class="font-semibold border-b border-theme-neutral-200"
            data-testid="LinkCollectionTable__row"
          >
            <td
              class="w-16 text-center"
            >
              <input
                class="sc-AxiKw fIRSaL"
                color="success"
                data-testid="LinkCollectionTable__row__option"
                type="radio"
              />
            </td>
            <td
              class="w-40 pb-6 pt-2"
              data-testid="LinkCollectionTable__row__type"
            >
              <input
                class="hidden"
                name="example[0].type"
                readonly=""
                value="facebook"
              />
              <span>
                Facebook
              </span>
            </td>
            <td
              class="pb-6 pt-2"
              data-testid="LinkCollectionTable__row__value"
            >
              <input
                class="focus:outline-none"
                name="example[0].value"
                readonly=""
                value="https://facebook.com"
              />
            </td>
            <td
              class="w-16 text-right pb-4"
            >
              <button
                class="sc-AxjAm bBBiEi"
                color="primary"
                data-testid="LinkCollectionTable__remove-link"
                type="button"
              >
                <div
                  class="sc-AxirZ iQaTP"
                  height="1em"
                  width="1em"
                >
                  <svg>
                    trash.svg
                  </svg>
                </div>
              </button>
            </td>
          </tr>
          <tr
            class="font-semibold border-b border-theme-neutral-200"
            data-testid="LinkCollectionTable__row"
          >
            <td
              class="w-16 text-center"
            >
              <input
                checked=""
                class="sc-AxiKw fIRSaL"
                color="success"
                data-testid="LinkCollectionTable__row__option"
                type="radio"
              />
            </td>
            <td
              class="w-40 py-6"
              data-testid="LinkCollectionTable__row__type"
            >
              <input
                class="hidden"
                name="example[1].type"
                readonly=""
                value="instagram"
              />
              <span>
                Instagram
              </span>
            </td>
            <td
              class="py-6"
              data-testid="LinkCollectionTable__row__value"
            >
              <input
                class="focus:outline-none"
                name="example[1].value"
                readonly=""
                value="https://instagram.com"
              />
            </td>
            <td
              class="w-16 text-right false"
            >
              <button
                class="sc-AxjAm bBBiEi"
                color="primary"
                data-testid="LinkCollectionTable__remove-link"
                type="button"
              >
                <div
                  class="sc-AxirZ iQaTP"
                  height="1em"
                  width="1em"
                >
                  <svg>
                    trash.svg
                  </svg>
                </div>
              </button>
            </td>
          </tr>
          <tr
            class="font-semibold border-b border-theme-neutral-200"
            data-testid="LinkCollectionTable__row"
          >
            <td
              class="w-16 text-center"
            />
            <td
              class="w-40 py-6"
              data-testid="LinkCollectionTable__row__type"
            >
              <input
                class="hidden"
                name="example[2].type"
                readonly=""
                value="twitter"
              />
              <span>
                Twitter
              </span>
            </td>
            <td
              class="py-6"
              data-testid="LinkCollectionTable__row__value"
            >
              <input
                class="focus:outline-none"
                name="example[2].value"
                readonly=""
                value="https://twitter.com"
              />
            </td>
            <td
              class="w-16 text-right false"
            >
              <button
                class="sc-AxjAm bBBiEi"
                color="primary"
                data-testid="LinkCollectionTable__remove-link"
                type="button"
              >
                <div
                  class="sc-AxirZ iQaTP"
                  height="1em"
                  width="1em"
                >
                  <svg>
                    trash.svg
                  </svg>
                </div>
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LinkCollectionTable With Options should show with options column 1`] = `
<DocumentFragment>
  <div
    data-testid="LinkCollectionTable"
  >
    <div
      class="sc-AxhCb tChBQ"
      role="table"
    >
      <table
        class="table-auto"
      >
        <thead>
          <tr
            role="row"
          >
            <th
              class="text-sm text-left select-none text-theme-neutral"
              colspan="1"
              data-testid="table__th--0"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex flex-inline align-top "
              >
                <div>
                  Avatar
                </div>
                <div
                  class="flex items-center ml-2 text-theme-color-neutral"
                  data-testid="table__Sort"
                >
                  <div
                    class="sc-AxirZ jMwzyA"
                    height="12"
                    width="15"
                  >
                    <svg>
                      sort.svg
                    </svg>
                  </div>
                </div>
              </div>
            </th>
            <th
              class="text-sm text-left select-none text-theme-neutral"
              colspan="1"
              data-testid="table__th--1"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex flex-inline align-top "
              >
                <div>
                  Name
                </div>
                <div
                  class="flex items-center ml-2 text-theme-color-neutral"
                  data-testid="table__Sort"
                >
                  <div
                    class="sc-AxirZ jMwzyA"
                    height="12"
                    width="15"
                  >
                    <svg>
                      sort.svg
                    </svg>
                  </div>
                </div>
              </div>
            </th>
            <th
              class="text-sm text-left select-none text-theme-neutral"
              colspan="1"
              data-testid="table__th--2"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex flex-inline align-top "
              >
                <div>
                  Link
                </div>
                <div
                  class="flex items-center ml-2 text-theme-color-neutral"
                  data-testid="table__Sort"
                >
                  <div
                    class="sc-AxirZ jMwzyA"
                    height="12"
                    width="15"
                  >
                    <svg>
                      sort.svg
                    </svg>
                  </div>
                </div>
              </div>
            </th>
            <th
              class="text-sm text-left select-none text-theme-neutral"
              colspan="1"
              data-testid="table__th--3"
              role="columnheader"
            >
              <div
                class="flex flex-inline align-top "
              >
                <div>
                   
                </div>
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            class="font-semibold border-b border-theme-neutral-200"
            data-testid="LinkCollectionTable__row"
          >
            <td
              class="w-16 text-center"
            >
              <input
                class="sc-AxiKw fIRSaL"
                color="success"
                data-testid="LinkCollectionTable__row__option"
                type="radio"
              />
            </td>
            <td
              class="w-40 pb-6 pt-2"
              data-testid="LinkCollectionTable__row__type"
            >
              <input
                class="hidden"
                name="example[0].type"
                readonly=""
                value="facebook"
              />
              <span>
                Facebook
              </span>
            </td>
            <td
              class="pb-6 pt-2"
              data-testid="LinkCollectionTable__row__value"
            >
              <input
                class="focus:outline-none"
                name="example[0].value"
                readonly=""
                value="https://facebook.com"
              />
            </td>
            <td
              class="w-16 text-right pb-4"
            >
              <button
                class="sc-AxjAm bBBiEi"
                color="primary"
                data-testid="LinkCollectionTable__remove-link"
                type="button"
              >
                <div
                  class="sc-AxirZ iQaTP"
                  height="1em"
                  width="1em"
                >
                  <svg>
                    trash.svg
                  </svg>
                </div>
              </button>
            </td>
          </tr>
          <tr
            class="font-semibold border-b border-theme-neutral-200"
            data-testid="LinkCollectionTable__row"
          >
            <td
              class="w-16 text-center"
            >
              <input
                class="sc-AxiKw fIRSaL"
                color="success"
                data-testid="LinkCollectionTable__row__option"
                type="radio"
              />
            </td>
            <td
              class="w-40 py-6"
              data-testid="LinkCollectionTable__row__type"
            >
              <input
                class="hidden"
                name="example[1].type"
                readonly=""
                value="instagram"
              />
              <span>
                Instagram
              </span>
            </td>
            <td
              class="py-6"
              data-testid="LinkCollectionTable__row__value"
            >
              <input
                class="focus:outline-none"
                name="example[1].value"
                readonly=""
                value="https://instagram.com"
              />
            </td>
            <td
              class="w-16 text-right false"
            >
              <button
                class="sc-AxjAm bBBiEi"
                color="primary"
                data-testid="LinkCollectionTable__remove-link"
                type="button"
              >
                <div
                  class="sc-AxirZ iQaTP"
                  height="1em"
                  width="1em"
                >
                  <svg>
                    trash.svg
                  </svg>
                </div>
              </button>
            </td>
          </tr>
          <tr
            class="font-semibold border-b border-theme-neutral-200"
            data-testid="LinkCollectionTable__row"
          >
            <td
              class="w-16 text-center"
            />
            <td
              class="w-40 py-6"
              data-testid="LinkCollectionTable__row__type"
            >
              <input
                class="hidden"
                name="example[2].type"
                readonly=""
                value="twitter"
              />
              <span>
                Twitter
              </span>
            </td>
            <td
              class="py-6"
              data-testid="LinkCollectionTable__row__value"
            >
              <input
                class="focus:outline-none"
                name="example[2].value"
                readonly=""
                value="https://twitter.com"
              />
            </td>
            <td
              class="w-16 text-right false"
            >
              <button
                class="sc-AxjAm bBBiEi"
                color="primary"
                data-testid="LinkCollectionTable__remove-link"
                type="button"
              >
                <div
                  class="sc-AxirZ iQaTP"
                  height="1em"
                  width="1em"
                >
                  <svg>
                    trash.svg
                  </svg>
                </div>
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LinkCollectionTable should delete link from DOM and form 1`] = `
<DocumentFragment>
  <div
    data-testid="LinkCollectionTable"
  >
    <div
      class="sc-AxhCb tChBQ"
      role="table"
    >
      <table
        class="table-auto"
      >
        <thead>
          <tr
            role="row"
          >
            <th
              class="text-sm text-left select-none text-theme-neutral"
              colspan="1"
              data-testid="table__th--0"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex flex-inline align-top "
              >
                <div>
                  Name
                </div>
                <div
                  class="flex items-center ml-2 text-theme-color-neutral"
                  data-testid="table__Sort"
                >
                  <div
                    class="sc-AxirZ jMwzyA"
                    height="12"
                    width="15"
                  >
                    <svg>
                      sort.svg
                    </svg>
                  </div>
                </div>
              </div>
            </th>
            <th
              class="text-sm text-left select-none text-theme-neutral"
              colspan="1"
              data-testid="table__th--1"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex flex-inline align-top "
              >
                <div>
                  Link
                </div>
                <div
                  class="flex items-center ml-2 text-theme-color-neutral"
                  data-testid="table__Sort"
                >
                  <div
                    class="sc-AxirZ jMwzyA"
                    height="12"
                    width="15"
                  >
                    <svg>
                      sort.svg
                    </svg>
                  </div>
                </div>
              </div>
            </th>
            <th
              class="text-sm text-left select-none text-theme-neutral"
              colspan="1"
              data-testid="table__th--2"
              role="columnheader"
            >
              <div
                class="flex flex-inline align-top "
              >
                <div>
                   
                </div>
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            class="font-semibold border-b border-theme-neutral-200"
            data-testid="LinkCollectionTable__row"
          >
            <td
              class="w-40 pb-6 pt-2"
              data-testid="LinkCollectionTable__row__type"
            >
              <input
                class="hidden"
                name="example[0].type"
                readonly=""
                value="instagram"
              />
              <span>
                Instagram
              </span>
            </td>
            <td
              class="pb-6 pt-2"
              data-testid="LinkCollectionTable__row__value"
            >
              <input
                class="focus:outline-none"
                name="example[0].value"
                readonly=""
                value="https://instagram.com"
              />
            </td>
            <td
              class="w-16 text-right pb-4"
            >
              <button
                class="sc-AxjAm bBBiEi"
                color="primary"
                data-testid="LinkCollectionTable__remove-link"
                type="button"
              >
                <div
                  class="sc-AxirZ iQaTP"
                  height="1em"
                  width="1em"
                >
                  <svg>
                    trash.svg
                  </svg>
                </div>
              </button>
            </td>
          </tr>
          <tr
            class="font-semibold border-b border-theme-neutral-200"
            data-testid="LinkCollectionTable__row"
          >
            <td
              class="w-40 py-6"
              data-testid="LinkCollectionTable__row__type"
            >
              <input
                class="hidden"
                name="example[1].type"
                readonly=""
                value="twitter"
              />
              <span>
                Twitter
              </span>
            </td>
            <td
              class="py-6"
              data-testid="LinkCollectionTable__row__value"
            >
              <input
                class="focus:outline-none"
                name="example[1].value"
                readonly=""
                value="https://twitter.com"
              />
            </td>
            <td
              class="w-16 text-right false"
            >
              <button
                class="sc-AxjAm bBBiEi"
                color="primary"
                data-testid="LinkCollectionTable__remove-link"
                type="button"
              >
                <div
                  class="sc-AxirZ iQaTP"
                  height="1em"
                  width="1em"
                >
                  <svg>
                    trash.svg
                  </svg>
                </div>
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LinkCollectionTable should display default links 1`] = `
<DocumentFragment>
  <div
    data-testid="LinkCollectionTable"
  >
    <div
      class="sc-AxhCb tChBQ"
      role="table"
    >
      <table
        class="table-auto"
      >
        <thead>
          <tr
            role="row"
          >
            <th
              class="text-sm text-left select-none text-theme-neutral"
              colspan="1"
              data-testid="table__th--0"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex flex-inline align-top "
              >
                <div>
                  Name
                </div>
                <div
                  class="flex items-center ml-2 text-theme-color-neutral"
                  data-testid="table__Sort"
                >
                  <div
                    class="sc-AxirZ jMwzyA"
                    height="12"
                    width="15"
                  >
                    <svg>
                      sort.svg
                    </svg>
                  </div>
                </div>
              </div>
            </th>
            <th
              class="text-sm text-left select-none text-theme-neutral"
              colspan="1"
              data-testid="table__th--1"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex flex-inline align-top "
              >
                <div>
                  Link
                </div>
                <div
                  class="flex items-center ml-2 text-theme-color-neutral"
                  data-testid="table__Sort"
                >
                  <div
                    class="sc-AxirZ jMwzyA"
                    height="12"
                    width="15"
                  >
                    <svg>
                      sort.svg
                    </svg>
                  </div>
                </div>
              </div>
            </th>
            <th
              class="text-sm text-left select-none text-theme-neutral"
              colspan="1"
              data-testid="table__th--2"
              role="columnheader"
            >
              <div
                class="flex flex-inline align-top "
              >
                <div>
                   
                </div>
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            class="font-semibold border-b border-theme-neutral-200"
            data-testid="LinkCollectionTable__row"
          >
            <td
              class="w-40 pb-6 pt-2"
              data-testid="LinkCollectionTable__row__type"
            >
              <input
                class="hidden"
                name="example[0].type"
                readonly=""
                value="facebook"
              />
              <span>
                Facebook
              </span>
            </td>
            <td
              class="pb-6 pt-2"
              data-testid="LinkCollectionTable__row__value"
            >
              <input
                class="focus:outline-none"
                name="example[0].value"
                readonly=""
                value="https://facebook.com"
              />
            </td>
            <td
              class="w-16 text-right pb-4"
            >
              <button
                class="sc-AxjAm bBBiEi"
                color="primary"
                data-testid="LinkCollectionTable__remove-link"
                type="button"
              >
                <div
                  class="sc-AxirZ iQaTP"
                  height="1em"
                  width="1em"
                >
                  <svg>
                    trash.svg
                  </svg>
                </div>
              </button>
            </td>
          </tr>
          <tr
            class="font-semibold border-b border-theme-neutral-200"
            data-testid="LinkCollectionTable__row"
          >
            <td
              class="w-40 py-6"
              data-testid="LinkCollectionTable__row__type"
            >
              <input
                class="hidden"
                name="example[1].type"
                readonly=""
                value="instagram"
              />
              <span>
                Instagram
              </span>
            </td>
            <td
              class="py-6"
              data-testid="LinkCollectionTable__row__value"
            >
              <input
                class="focus:outline-none"
                name="example[1].value"
                readonly=""
                value="https://instagram.com"
              />
            </td>
            <td
              class="w-16 text-right false"
            >
              <button
                class="sc-AxjAm bBBiEi"
                color="primary"
                data-testid="LinkCollectionTable__remove-link"
                type="button"
              >
                <div
                  class="sc-AxirZ iQaTP"
                  height="1em"
                  width="1em"
                >
                  <svg>
                    trash.svg
                  </svg>
                </div>
              </button>
            </td>
          </tr>
          <tr
            class="font-semibold border-b border-theme-neutral-200"
            data-testid="LinkCollectionTable__row"
          >
            <td
              class="w-40 py-6"
              data-testid="LinkCollectionTable__row__type"
            >
              <input
                class="hidden"
                name="example[2].type"
                readonly=""
                value="twitter"
              />
              <span>
                Twitter
              </span>
            </td>
            <td
              class="py-6"
              data-testid="LinkCollectionTable__row__value"
            >
              <input
                class="focus:outline-none"
                name="example[2].value"
                readonly=""
                value="https://twitter.com"
              />
            </td>
            <td
              class="w-16 text-right false"
            >
              <button
                class="sc-AxjAm bBBiEi"
                color="primary"
                data-testid="LinkCollectionTable__remove-link"
                type="button"
              >
                <div
                  class="sc-AxirZ iQaTP"
                  height="1em"
                  width="1em"
                >
                  <svg>
                    trash.svg
                  </svg>
                </div>
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
